docs/proposals/cluster-autoscaling.md

```diff
@@ -0,0 +1,314 @@
+
+In version 1.2 we will provide “small” KAC. Its responsibility will be mainly limited to reading the config and updating Google
+Cloud Autoscaler accordingly. It will also observe changes in MIG and generate appropriate kubernetes events.
+
+```
+                          +--MIG-----------------------------+
+                          |                                  |
+                          | +--------+ +--------+ +--------+ |
+                          | |        | |        | |        | |
+                          | | Node1  | | Node2  | | NodeX  | |
+                          | |        | |        | |        | +<--+
+                          | +--------+ +--------+ +--------+ |   |
+            +---------+   | |cAdvisor| |cAdvisor| |cAdvisor| |   |
+            |  Config |   | +--------+ +--------+ +----+---+ |   |
+            +---------+   |     |          |           |     |   |
+                 |        +-----|----------|-----------|-----+   |
+                 |              |          |           |         |
+            +----v----+         |      +---v----+      |         |
+            |   KAC   |         +----> |Heapster| <----+         |
+            +---------+                +---+----+                |
+                 |                         |                     |
+ Kubernetes      |                         |                     |
+-----------------|-------------------------|---------------------|------
+  Google         |                         |                     |
+                 |                         |                     |
+                 |                  +------v-----+       +-------+------+
+                 |                  |Google Cloud|       |MIG Definition|
+                 |                  | Monitoring |       +-------^------+
+                 |                  +------------+               |
+                 |                         |                     |
+                 |                         |                     |
+                 |                  +------v-----+               |
+                 +----------------->|Google Cloud+---------------+
+                                    | Autoscaler |
+                                    +------------+
+```
+
+
+We will also configure GCE Autoscaler so that it deletes nodes  in a non-random way. There are couple strategies for killing to be considered:
+
+* Node with the fewest number of pods.
+* Node with the fewest number of non-replicated pods.
+* Node with the lowest load.
+
+GCA will know about our preferences from the metrics that would go to GCM the similar way as cpu-usage (it will kill nodes starting from the one with the lowest metric). As for 1.2 the corresponding API may not be ready we will use a predefined metric name and suffix for GCA name.
+
+We will use MIG shutdown script to support pod graceful terminate in case of node deletion.
```

Comment:
s/terminate/termination
Might be worth explaining how this will work -- will you do something like what is described in #3885 ?